src/components/activity/ActivityDirectivesTable.svelte

```diff
@@ -138,13 +174,36 @@
     return activityDirective.id;
   }
 
+  function getSelectedActivityDirectives(): ActivityDirective[] {
+    const directives: ActivityDirective[] = [];
+    bulkSelectedActivityDirectiveIds.forEach(id => {
+      const found = activityDirectives.find(item => item.id === id);
+      if (found !== null && found !== undefined) {
+        directives.push(found);
+      }
+    });
+    return directives;
+  }
+
   function scrollTimelineToActivityDirective() {
     const directiveId = bulkSelectedActivityDirectiveIds.length > 0 && bulkSelectedActivityDirectiveIds[0];
     const directive = activityDirectives.find(item => item.id === directiveId) ?? null;
     if (directive?.start_time_ms !== undefined && directive?.start_time_ms !== null) {
       dispatch('scrollTimelineToTime', directive.start_time_ms);
     }
   }
+
+  function copyActivityDirectives({ detail: activities }: CustomEvent<ActivityDirective[]>) {
+    plan !== null && copyActivityDirectivesToClipboard(plan, activities);
+  }
+
+  function canPasteActivityDirectives(): boolean {
+    return plan !== null && hasCreatePermission && canPasteActivityDirectivesFromClipboard(plan);
+  }
+
+  function pasteActivityDirectives() {
+    plan !== null && pasteActivityDirectivesFromClipboard(plan, user);
```

Comment:
I think this needs to check the permission as well in case the user uses the keyboard shortcut.

Reply:
I think that is already handled by the resulting `cloneActivityDirectives` effect?

Reply:
The permission in `cloneActivityDirectives` is only the query permission. There's not enough resolution in query permissions because it just determines if the user's role has access to a certain query. What I think we might need here is to handle the feature permission (i.e. if they are a owner/collaborator of the target plan and whatever nuances need to be taken into account)